### PR TITLE
Fixing jenkinsfile for new k8s slaves

### PR DESCRIPTION
**Pull request by michelsciortino**
This PR has been created to fix Jenkins pipeline for new k8s slaves:

- NEW: `CI/extract_from_docker_image.sh` script extracts all the binaries, headers and libraries
  from docker image to the slave file-system                  
- NEW: `CI/clean_slave.sh` script cleans the slave's file-system deleting all the file extracted by `CI/extract_from_docker_image.sh`
- Jenkins test stages now run `CI/extract_from_docker_image.sh`, execute tests and run `CI/clean_slave.sh`
- `pcn-pbforwarder` tests have been modified as they were using a wrong grep regex to obtain the mac addresses of the namespaces
      `'HWaddr \K.*$'` -> `'ether \K[a-fA-F0-9:]{17}|[a-fA-F0-9]{12}$'`
- all the `netcat` timeouts have been increased from 2 to 5 seconds in `pcn-nat`'s and `pcn-iptables`'s tests

Jenkins pipeline is partially fixed as there is still a failing test of `pcn-iptables` (`local_test_interfaces1.sh`): a new Issue will be opened soon to report this.

`Clean Instance Tests` and `Same Instance Tests`  now work and can be used as mandatory checks for new PRs.

### CODE REVIEW

File: CI/JenkinsfileTestCleanInstance [hunk 1]
```diff
@@ -215,17 +215,45 @@ pipeline {
...
                                     docker system prune --all --force
                                     # Launch the polycubed daemon and extract the polycubectl
                                     # /proc is mounted under /host/proc to have a complete visibility of the namespaces created outside of the container
-                                    docker run -d --name polycubed --rm --privileged --pid=host --cap-add ALL --network host -v /proc:/host/proc -v /lib/modules:/lib/modules:ro -v /var/run/netns/:/var/run/netns:rw -v /usr/src:/usr/src:ro -v /etc/localtime:/etc/localtime:ro polycubebot/${image_name}-default:${image_tag}
-                                    mkdir bin
-                                    docker cp polycubed:/usr/local/bin/polycubectl ./bin/
+                                    docker run -d --name \$container --rm --privileged --pid=host --cap-add ALL --network host -v /proc:/host/proc -v /lib/modules:/lib/modules:ro -v /var/run/netns/:/var/run/netns:rw -v /usr/src:/usr/src:ro -v /etc/localtime:/etc/localtime:ro polycubebot/${image_name}-default:${image_tag}
+                                    
+                                    # Copying binaries
```
Comment (frisso) [inline CI/JenkinsfileTestCleanInstance]: Sorry for the silly question: is there any way to define this code ones, such as with a function? It looks we have a lot of duplicated code here.

**Comment by frisso** [comment]
@michelsciortino Two comments.
1) I believe this is still a work in progress, given the number of commits you have. Please mark as "WIP" so that everybody is aware of its status. Obviously, before finalizing the PR, we should perform a squash commit.
2) The current code is out-of-data with the base branch. Remember to perform a git rebase before finalizing the PR.
Thanks!

**Comment by michelsciortino** [comment]
@here This PR is ready to review, please check it out

**Comment by palexster** [comment]
Hi @michelsciortino, thanks for this! 
The remaining point is to move flaky tests out of the CI steps in next PRs in order to avoid false positives during builds. This would make the activation of new mandatory checks less painful for everybody.

**Pull request merged**